src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -211,10 +215,29 @@ static void send_sntp_sync(struct timeval *tv)
     END_WITH_STACK_HEAP(heap, driver_data->global);
 }
 
+// param should be pointer to malloc'd destination to copy device hostname
+// return ok atom or error as atom
+static term get_default_device_name(char *name, GlobalContext *global)
```

Comment:
We might rework this function to be just a helper for writing to the hostname buffer.
Such as `static bool write_default_device_name(char *out, size_t out_len);`
The caller may decide how to handle the error.